src/main/java/bio/terra/workspace/service/workspace/flight/CreateWorkspaceStep.java

```diff
@@ -55,9 +58,14 @@ public StepResult doStep(FlightContext flightContext) throws RetryException {
   public StepResult undoStep(FlightContext flightContext) {
     FlightMap inputMap = flightContext.getInputParameters();
     FlightMap workingMap = flightContext.getWorkingMap();
-    if (workingMap.get(CREATE_WORKSPACE_COMPLETED_KEY, Boolean.class)) {
-      UUID workspaceId = inputMap.get(WorkspaceFlightMapKeys.WORKSPACE_ID, UUID.class);
+    UUID workspaceId = inputMap.get(WorkspaceFlightMapKeys.WORKSPACE_ID, UUID.class);
+    if (workingMap.get(CREATE_WORKSPACE_COMPLETED_KEY, Boolean.class) != null) {
```

Comment:
same quesiton for this.

Reply:
Same as the other step, that gets set in the `do` step as a way of confirming that a workspace was created in this flight.

Reply:
ditto, I don't think this is sufficient. Let's do locking or unique operation ids.